### PR TITLE
fix(deployments): remove erroneous creation of work queue

### PR DESCRIPTION
**Pull request by joelluijmes**
Creation of work queue and pool is already handled upstream at src/prefect/server/api/deployments.py.

When creating a deployment with different work pool, this `_ensure_work_queue_exists` causes erroneous work queues to be created in the default work pool.

---

## Reproduction Example

Below is a short example of the current behaviour. 

### 1. List the current work queues:

```bash
❯ prefect work-queue ls 
                                        Work Queues                                        
┏━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━┓
┃ Name    ┃ Pool               ┃                                   ID ┃ Concurrency Limit ┃
┡━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━┩
│ local   │ default-agent-pool │ a37973a4-6674-4d22-8bfe-0f04ede0ed9f │ None              │
│ default │ default-agent-pool │ 3d4f276e-8d05-48a9-a3cd-bfc37dfa7dbb │ None              │
└─────────┴────────────────────┴──────────────────────────────────────┴───────────────────┘
                                (**) denotes a paused queue                                
```

### 2. Create a deployment with configured work pool and queue

```python
import asyncio

from prefect import flow
from prefect.deployments import Deployment
from prefect.client.orchestration import get_client

@flow
def my_flow(arg):
    print(arg)


async def main():
    try:
        async with get_client() as client:
            await client.create_work_pool(name="my_pool")
    except:
        pass

    await Deployment.build_from_flow(
        my_flow,
        "prod",
        parameters=dict(arg="hello"),
        apply=True,
        load_existing=False,
        work_queue_name="my_queue",
        work_pool_name="my_pool",
    )


if __name__ == "__main__":
    asyncio.run(main())
```

### 3. Listing work queues again shows `my_queue` also created in the default `default-agent-pool`

```bash
❯ prefect work-queue ls 
                                        Work Queues                                         
┏━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━┓
┃ Name     ┃ Pool               ┃                                   ID ┃ Concurrency Limit ┃
┡━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━┩
│ my_queue │ default-agent-pool │ d3c566d9-2d12-4d73-9581-e15431c0ef87 │ None              │
│ my_queue │ my_pool            │ d82b0291-db8a-4ea8-ac9e-1f8f7e38d62a │ None              │
│ default  │ my_pool            │ da373e86-b272-4c13-8b90-5a7b0f9bd020 │ None              │
│ local    │ default-agent-pool │ a37973a4-6674-4d22-8bfe-0f04ede0ed9f │ None              │
│ default  │ default-agent-pool │ 3d4f276e-8d05-48a9-a3cd-bfc37dfa7dbb │ None              │
└──────────┴────────────────────┴──────────────────────────────────────┴───────────────────┘
```

### Checklist

- [x] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [ ] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`, `docs`.

### CODE REVIEW

File: src/prefect/server/models/deployments.py [hunk 1]
```diff
@@ -106,11 +106,6 @@ async def create_deployment(
     result = await session.execute(query)
     model = result.scalar()
 
-    if model.work_queue_name:
-        await models.work_queues._ensure_work_queue_exists(
-            session=session, name=model.work_queue_name, db=db
-        )
```
Comment (desertaxle) [inline src/prefect/server/models/deployments.py:112]: I see that there's a call like this in `update_deployment` in this file as well which should also be removed. Would you be willing to remove it as part of this PR?

**Review by desertaxle — COMMENTED** [review]
Thanks for reporting this bug and implementing a fix @joelluijmes! In addition to the comment about removing another call of `_ensure_work_queue_exists` in `update_deployment, could you add a test to ensure that duplicate queues are no longer created in the default agent pool?

File: src/prefect/server/models/deployments.py (hunk 1 above)
Comment (joelluijmes) [reply src/prefect/server/models/deployments.py:112]: Ah yes, I missed that one. Removed the call there as well!

**Comment by joelluijmes** [comment]
Sure, if you want I can add such a test. But to me it feels bit odd to test for something that shouldn't happen in the first place.

Regardless, I tried testing it by listing the queue_names after the update call in a test, but the to be duplicated queue is already there. I think the database might not be cleared between test runs? Well at least not when running it as `pytest ...`.

```
queues = await models.work_queues.read_work_queues(session=session)
print([(x.work_pool.name, x.name) for x in queues])
```

Let me know if you have other suggestions how I can create such tests :)

**Comment by desertaxle** [comment]
Writing a test that queues aren't duplicated across work queues will prevent regression and ensure that the requirement continues to be met. Depending on the work queue name that you chose, it might have already been created by a fixture. You could append a UUID on the end of your work queue name to ensure the work queue name will be unique.

File: tests/server/models/test_deployments.py [hunk 2]
```diff
@@ -34,30 +34,36 @@ async def test_create_deployment_succeeds(
...
         await models.deployments.create_deployment(
             session=session,
             deployment=schemas.core.Deployment(
-                name="d1", work_queue_name="wq-1", flow_id=flow.id, manifest_path=""
+                name="d1",
+                work_queue_name=work_queue.name,
+                flow_id=flow.id,
+                manifest_path="",
             ),
         )
```
[the remark this answers is not in the document]
Comment (joelluijmes) [reply tests/server/models/test_deployments.py:54]: This is obviously shown at `test_creating_a_deployment_does_not_create_work_queue`

**Review by desertaxle — APPROVED** [review]
LGMT!
